source/extensions/filters/common/expr/context.cc

```diff
@@ -139,6 +140,20 @@ absl::optional<CelValue> ResponseWrapper::operator[](CelValue key) const {
     return CelValue::CreateMap(&trailers_);
   } else if (value == Flags) {
     return CelValue::CreateInt64(info_.responseFlags());
+  } else if (value == GrpcStatus) {
+    for (const auto wrapper : ((HeadersWrapper[]){trailers_, headers_})) {
+      if (wrapper.value_ != nullptr) {
+        const auto& optional_status = Grpc::Common::getGrpcStatus(*(wrapper.value_));
+        if (optional_status.has_value()) {
+          return CelValue::CreateInt64(optional_status.value());
+        }
+      }
+    }
+    auto code = info_.responseCode();
+    if (code.has_value()) {
+      return CelValue::CreateInt64(Grpc::Utility::httpToGrpcStatus(code.value()));
+    }
+    return CelValue::CreateInt64(Grpc::Status::WellKnownGrpcStatus::Unknown);
```

Comment:
I don't think that returning `grpc-status: 2` (unknown) when it wasn't present in the response is a correct thing to do.

Reply:
@PiotrSikora I was trying to match: https://github.com/envoyproxy/envoy/blob/5248a4fb7d4c2a3d1fa151f944d3a63f6b7a06cf/source/common/access_log/access_log_impl.cc#L231-L261.

If not `UNKNOWN`, what status would you expect as a return value?

Reply:
@douglas-reid fair point wrt being consistent with the access loggers, but I don't think any status should be returned as a value if there was no status sent on the wire...

Can't the return value by `{}`, `absl::nullopt`, or something else in that style?

Reply:
It can definitely return `{}`. That was the way I had it originally.

I guess we need to decide on exactly what to do.

BTW, it looks like choosing "Unknown" has some prior art:

https://github.com/grpc/grpc-go/blob/master/internal/transport/http_util.go#L281-L288

Reply:
I think it should be `{}`. There is quite a difference between missing field and `grpc-status: 2`.

Reply:
I think it should be `{}`. Otherwise, we run the risk of confusing the consumer for a non-gRPC request.